### PR TITLE
Interactive Selection Implementation

### PR DESCRIPTION
**Pull request by zFernand0**
Added support for interactive selection

Use cases:
- [x] `zowe jobs view interactive-jobs`
- [ ] Download the selected spool file
- [ ] List and view dataset/members
- [ ] Download the selected member/dataset
- [ ] List and view USS files
- [ ] Download the selected USS file
- ... more

Design consideration:
1. Smart Interactions:
    - Move interactive selection back one/two level/s to `zowe jobs interact` or `zowe jobs --interactive`
    - From here the user will be presented with a list of jobs (based on `--owner` and `--prefix`)
    - Then the user can select the job (up/down arrow keys) and select which action to perform (left/right arrow keys)
    - The applicable actions to be selected are dictated by the verbs under `zowe jobs` 
      - View
      - Download
      - Delete
      - Cancel
    - Same could apply to `zowe files`, `zowe workflows` ...
2. Interactive CLI
    - Implement Interactive Menus at any level, e.g.  `zowe --interactive`, `zowe tso --interactive`
    - From here the user will be presented with a list of available commands at that level to select.
    - The user will continue selecting commands `group`s until they reach the last level (`command`)
    - For example: `zowe --interactive`
      - List:
        ```
          zos-tso
          zos-files
          ...
          zosmf 
        ```
      - Select: `zosmf`
        New list:
        ```
          check
          list
        ```
      - Select: `check`
        New list:
        ```
          status
        ```
      - Select: `status`
      - Command issued `zowe zosmf check status`

TODO:
- [ ] `Ctrl + C` or `Esc` to cancel out of the Interactive Mode (not working for daemon mode)
- [ ] Implement single row menu (left/right arrow keys)
- [ ] `Backspace` for goin back to the previous action 😋

### CODE REVIEW

File: zowex/src/main.rs [hunk 1]
```diff
@@ -158,6 +162,36 @@ fn run_zowe_command(mut args: String, port_string: &str) -> std::io::Result<()>
...
+
+                            // append response to header
+                            let mut full_reply = X_ZOWE_DAEMON_REPLY.to_owned();
+                            full_reply.push_str(&selected_entry.to_string());
+
+                            // adjust our state that prompting has been resolved
+                            headers.insert(X_ZOWE_DAEMON_INTERACTIVE.to_string(), 0i32);
+
+                            // write response
+                            stream_clone.write(full_reply.as_bytes()).unwrap();
```
Comment (dkelosky) [inline zowex/src/main.rs:193]: you're a rust master now 😉

File: zowex/src/main.rs [hunk 2]
```diff
@@ -237,6 +272,7 @@ fn get_beg(buf: &str) -> Vec<String> {
 
     // if this line is long enough to contain headers
     if ss.len() >= 2 {
+        // TODO(???) - Is it safe to assume `x-` is for daemon headers only? What if the data contains `x-`?
```
Comment (dkelosky) [inline zowex/src/main.rs:275]: this was my failure, i see you opened an issue for it.  we'll fix it someday 😄

File: zowex/src/main.rs (hunk 1 above)
Comment (zFernand0) [reply zowex/src/main.rs:193]: I wish! 😋

**Comment by zFernand0** [comment]
NodeJS Example
nodejs-example

Daemon Example
daemon-example

**Comment by zFernand0** [comment]
NodeJS Update:
nodejs-jobs-interact

Daemon Update (coming soon)

**Comment by zFernand0** [comment]
I believe that there are too many changes required for this PR to become ready for review.
It might be best to close this and create a new one when/if there's any interest.
Also, this could be moved to a plug-in that introduces interaction and animations to the terminal

**Pull request closed without merging**